### DISCUSSION TITLE
Improving live streaming performance

### DISCUSSION THREAD

**Discussion by jeremyhiggs** [dgreif/ring#910] [Ideas]
I tried out Scrypted for a few days and found it quite lacking compared to Homebridge. One of the compelling features (in theory) was quicker loading of snapshots and streams, which I believe is achieved by constantly buffering the stream.

Is that feasible for homebridge-ring? Otherwise, how can the responsiveness of the livestream and snapshots be improved?

**Comment by dgreif** [comment]
Constantly streaming isn't really a viable option because the streams all go through Rings servers (with Ring Edge as the exception).  I have a beta version out currently that should have faster stream startup speeds in general, as well as support for Ring Edge.  

On the snapshot front, the limitations come mainly from the Ring side, particularly on battery cameras.  The main piece that would help solve this is getting wired into push notifications from the Ring API, but that's an area that is hard to crack and I have not gotten any support from my official Ring contacts on this front. I could show _stale_ snapshots to help combat some of these issues, but I feel like that would also be considered a bug.  See https://github.com/dgreif/ring/wiki/Snapshot-Limitations for more details if you haven't already
